app/services/mentee_application_transition_service.rb

```diff
@@ -35,34 +35,28 @@ class InvalidTransitionError < StandardError; end
     }
   }.freeze
 
-  def promote!(application:, reviewer:, note: nil)
+  def call(application:, reviewer:, action:, note: nil)
     status = application.current_status.to_sym
-    raise InvalidTransitionError unless STATUS_TRANSITION_MAPPING[status][:valid_transitions].include? :promote
-    promote_transition = STATUS_TRANSITION_MAPPING[status][:promote_transition]
-    application.mentee_application_states.create!(status: promote_transition, reviewer:, note:)
-    invoke_side_effects(application, promote_transition, reviewer)
+    # enforce that the action is cast to a symbol before performing guard clause
+    action = action.to_sym
+    raise InvalidTransitionError unless STATUS_TRANSITION_MAPPING[status][:valid_transitions].include? action
```

Comment:
Since you've created `.valid_transitions`, I think it'd be nice to use it with this guard clause.

Reply:
Good call out! Will update :)